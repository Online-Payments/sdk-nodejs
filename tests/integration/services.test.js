const sdkInstance = require('./__fixture__');
const config = require('./__fixture__/config');

describe('Services', () => {
  describe('Promises', () => {
    test('it can test a connection', async () => {
      const sdkResponse = await sdkInstance.services.testConnection(config.merchantId, {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });
  });

  describe('Callback', () => {
    test('it can test a connection', (done) => {
      sdkInstance.services.testConnection(config.merchantId, {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });
  });
});
