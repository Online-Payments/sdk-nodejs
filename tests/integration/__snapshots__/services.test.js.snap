// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Callback it can test a connection 1`] = `
Object {
  "body": Object {
    "result": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Services Promises it can test a connection 1`] = `
Object {
  "body": Object {
    "result": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;
